src/umpire/op/HostReallocateOperation.cpp

```diff
@@ -36,9 +36,11 @@ void HostReallocateOperation::transform(
   ResourceManager::getInstance().deregisterAllocation(src_ptr);
   *dst_ptr = ::realloc(src_ptr, length);
 
+  // Ensure the realloc was successful
+  UMPIRE_ASSERT(*dst_ptr);
+
```

Comment:
This one should actually be an UMPIRE_ERROR since we want for the allocation failure to get back to the user so they can recover.